tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -760,6 +789,69 @@ public void ViewAreProhibitedTest([Values(false, true)] bool async)
             }
         }
 
+        // NOTE: this test is not presented in the prose tests
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void UnsupportedPlatformsTests(
+            [Values("local", "aws", "azure", "gcp")] string kmsProvider,
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            using (var clientEncrypted = ConfigureClientEncrypted())
+            using (var clientEncryption = ConfigureClientEncryption(clientEncrypted.Wrapped as MongoClient))
+            {
+                Guid? dataKeyId = null;
+                var dataKeyOptions = CreateDataKeyOptions(kmsProvider);
+                var exception = Record.Exception(() => dataKeyId = CreateDataKey(clientEncryption, kmsProvider, dataKeyOptions, async));
+                AsserResult(exception);
+            }
+
+            void AsserResult(Exception ex)
+            {
+                var isLinux = RequirePlatform.GetCurrentOperatingSystem() == SupportedOperatingSystem.Linux;
+
+                switch (kmsProvider)
+                {
+                    // all crypto hooks fail on linux with .netstandard1.5
+                    case var _ when isLinux && ContainsCurrentTargetFramework(SupportedTargetFramework.NetCoreApp11):
+                        {
+                            var errorMessage = AssertExceptionTypesAndReturnErrorMessage<PlatformNotSupportedException>(ex);
+                            errorMessage.Should().Be($"Field-level encryption is not supported on Linux with .netstandard1.5.");
+                        }
+                        break;
+                    case "gcp" when isLinux && ContainsCurrentTargetFramework(SupportedTargetFramework.NetCoreApp21):
+                        {
+                            var errorMessage = AssertExceptionTypesAndReturnErrorMessage<CryptException>(ex);
+                            errorMessage.Should().Be("error constructing KMS message: Failed to create GCP oauth request signature");
+                        }
+                        break;
+                    default:
+                        ex.Should().BeNull(); // the rest of cases should not throw
+                        break;
+                }
+            }
+
+            string AssertExceptionTypesAndReturnErrorMessage<TException>(Exception ex) where TException : Exception
+            {
+                var e = ex.Should().BeOfType<MongoEncryptionException>().Subject;
+                return e.InnerException.Should().BeOfType<TException>().Subject.Message;
+            }
+
+            bool ContainsCurrentTargetFramework(params SupportedTargetFramework[] supportedTargetFrameworks)
```

Comment:
No need for `params` here. Suggest more simple version:

    bool CurrentTargetFrameworkIs(SupportedTargetFramework supportedTargetFramework)
    {
        var currentTargetFramework = RequirePlatform.GetCurrentTargetFramework();
        return supportedTargetFramework == currentTargetFramework;
    }

Reply:
done